packages/e2e-tests/specs/experiments/navigation-editor.test.js

```diff
@@ -679,4 +680,143 @@ describe.skip( 'Navigation editor', () => {
+				await createMenu( { name: menuName }, menuItemsFixture );
+				await visitNavigationEditor();
+				await setBrowserViewport( viewport );
+				// Wait for the header to show the menu name.
+				await page.waitForXPath(
+					`//h2[contains(text(), "${ menuName }")]`
+				);
+				if ( viewport === 'small' ) {
+					const openSettingsSidebar = await page.waitForXPath(
+						'//button[@aria-label="Settings"][@aria-expanded="false"]'
+					);
+					await openSettingsSidebar.click();
+				}
+
+				const deleteMenuButton = await page.waitForXPath(
+					'//button[text()="Delete menu"]'
+				);
+				await deleteMenuButton.click();
+
+				const confirmButton = await page.waitForXPath(
+					'//*[@role="dialog"]//button[text()="OK"]'
+				);
+				await confirmButton.click();
+
+				await page.waitForXPath(
+					`//*[@role="button"][@aria-label="Dismiss this notice"]//*[text()='"${ menuName }" menu has been deleted']`
+				);
+
+				// If the "Create your first menu" prompt appears, we know there are no remaining menus,
+				// so our test menu must have been deleted successfully.
+				const createFirstMenuPrompt = await page.waitForXPath(
+					'//h3[.="Create your first menu"]',
+					{
+						visible: true,
+					}
+				);
+				const noMenusRemaining = createFirstMenuPrompt ? true : false;
+				expect( noMenusRemaining ).toBe( true );
+			}
+		);
+
+		it.each( [ 'large', 'small' ] )(
+			`should delete menu when confirmation is confirmed and there are other existing menus and the viewport is %s`,
+			async () => {
+				const menuName = 'Menu delete test';
+				await createMenu( { name: menuName }, menuItemsFixture );
+				await createMenu(
+					{ name: `${ menuName } 2` },
+					menuItemsFixture
+				);
+				await visitNavigationEditor();
+				// Wait for the header to show the menu name.
+				await page.waitForXPath(
+					`//h2[contains(text(), "${ menuName }")]`
+				);
+
+				const deleteMenuButton = await page.waitForXPath(
+					'//button[text()="Delete menu"]'
+				);
+				await deleteMenuButton.click();
+
+				const confirmButton = await page.waitForXPath(
+					'//*[@role="dialog"]//button[text()="OK"]'
+				);
+				await confirmButton.click();
+
+				await page.waitForXPath(
+					`//*[@role="button"][@aria-label="Dismiss this notice"]//*[text()='"${ menuName }" menu has been deleted']`
+				);
+
+				const menuActionsDropdown = await page.waitForXPath(
+					'//*[@class="edit-navigation-menu-actions"]//button[@aria-expanded="false"]'
+				);
+
+				await menuActionsDropdown.click();
+
+				const menuElementHandles = await page.$x(
+					'//*[@role="group"]//*[@role="menuitemradio"]/span'
+				);
+				const existingMenus = [];
+				for ( const elem of menuElementHandles ) {
+					const elemName = await elem.getProperty( 'textContent' );
+					const existingMenuName = await elemName.jsonValue();
+					existingMenus.push( existingMenuName );
+				}
+				expect( existingMenus.includes( `${ menuName }` ) ).toBe(
+					false
+				);
```

Comment:
This part feels very "implementation details"-oriented.. maybe a better way to test for this is to (if I understand the purpose of this code):

- Create both menu1 and menu2
- Check that there is a dropdown entry (menuitemradio) for both menus
- Delete menu2
- Check that there is still a dropdown entry for the menu1
- Check for menu2 entry, expect the result to be `null` (or whatever the result should be when an element can't be found in the page)

Reply:
Good suggestion. I pushed 26fbcc5. Unfortunately `waitForXPath()` doesn't return anything when it fails to find an element, it just times out, throws an error and stops the test.

`$x()` [is meant to return an array of matching ElementHandles](https://pptr.dev/#?product=Puppeteer&version=v13.4.0&show=api-pagexexpression), which I could then try to call `toContain()` against... but I had some trouble implementing it so I wound up falling back to `waitForXPath()` and parsing `TimeOut` errors as `null` to serve as an indicator that the element isn't present. What do you think?

Reply:
I'm not an expert about puppetteer APIs, but overall it seems a good enough approach.

My only comment, at this point, is regarding the verbosity of these checks — have you searched if there's already a shared util for performing such a check?

Alternatively, we could simply refactor the logic into a `getMenuItem( menuItemName )` function, something like:

```js
const getMenuItem = ( menuItemName ) =>
  await page
    .waitForXPath(
      `//*[@role="group"]//*[@role="menuitemradio"]/span[text()="${ menuItemName }"]`
    )
    .catch( ( error ) => {
      if ( error.name !== 'TimeoutError' ) {
        throw error;
      } else {
        return null;
      }
    } );
```

Reply:
I haven't seen any similar shared utilities to easily check for the existence of an element, but I like your refactoring suggestion, thank you! pushed f6a2f44 :)